pygbm/splitting.py

```diff
@@ -349,19 +353,19 @@ def _find_histogram_split(context, feature_idx, sample_indices):
     if root_node:
         if context.constant_hessian:
             histogram = _build_histogram_root_no_hessian(
-                context.n_bins, binned_feature, ordered_gradients)
+                context.max_bins, binned_feature, ordered_gradients)
```

Comment:
Actually we should also pass `n_bins_per_feature[feature_idx]` to all histogram computation routines.

`max_bins` is used to create histograms with a uniform shape, and `n_bins_per_feature[feature_idx]` is used to iterate over the relevant bins only